cmd/ls/ls.go

```diff
@@ -43,12 +45,30 @@ func formatFileName(name string) string {
 	return name
 }
 
+func lookupIds(uid, gid string) (string, string) {
+	u, err := user.LookupId(uid)
+	if err != nil {
+		fatal(err)
+	}
+	g, err := user.LookupGroupId(gid)
+	if err != nil {
+		fatal(err)
+	}
+	return u.Username, g.Name
+}
+
 func printFileList(fileInfo os.FileInfo) string {
+	stat := fileInfo.Sys().(*syscall.Stat_t)
```

Comment:
you're right. Another problem is the missing check for assertion success.

About multiple operating systems implementations, this code is fine for unix in general. Requires move this code to some file _unix.go, or add the proper build flags.